src/applications/personalization/profile/components/ProfileSubNavItems.jsx

```diff
@@ -21,31 +27,79 @@ function ProfileSubNavItems({ routes, isLOA3, isInMVI, clickHandler = null }) {
     // mvi check
     return !(route.requiresMVI && !isInMVI);
   });
-  const recordNavUserEvent = () => {
+
+  const recordNavUserEvent = e => {
     recordEvent({
       event: 'nav-sidenav',
     });
-    if (clickHandler) {
-      clickHandler();
-    }
+    const { href } = e.detail;
+    history.push(href);
   };
+
+  const isActive = path => (pathname === path ? true : undefined);
+
   return (
-    <ul className="vads-u-margin-top--0">
+    <VaSidenav
+      header="Profile"
+      icon-background-color="vads-color-primary"
+      icon-name="account_circle"
+      role="navigation"
+    >
       {filteredRoutes.map(route => {
-        return (
-          <li key={route.path}>
-            <NavLink
-              activeClassName="is-active"
-              exact
-              to={route.path}
-              onClick={recordNavUserEvent}
+        // Checks if route should be rendered inside a submenu by looking for a subnavParent
+        // If route has a subnavParent, checks for all other routes with the same subnavParent
+        // If there are 2 or more routes with the same subnavParent, render them inside a submenu
+        // If route is not the first child of a subnavParent, skip because it's rendered elsewhere
+        // If there is only 1 route for a subnavParent, render as individual route
+        if (route.subnavParent) {
+          const subnavChildren = filteredRoutes.filter(
+            subnavRoute => subnavRoute.subnavParent === route.subnavParent,
+          );
+          const hasSubnavChildren = subnavChildren.length > 1;
+          const isFirstChild = subnavChildren[0]?.name === route.name;
+          if (!isFirstChild) return null;
+          if (!hasSubnavChildren) {
```

Comment:
Couldn't we just invert the condition to `if (hasSubnavChildren)` and let the `!hasSubnavChildren` case fall through to line 91?

Reply:
The return needs to happen inside the `if (route.subnavParent)` check. `if (hasSubnavChildren)` falls to line 91, a submenu won't be rendered.